src/pager.c

```diff
@@ -1429,6 +1430,20 @@ char *supplemental_name;
         if (*dbase_str == ' ')
             ++dbase_str;
     }
+
+    /* remove material type */
+    int l;
+    for (i = 1; i <= NUM_MATERIAL_TYPES; i++) {
+        l = strlen(materialnm[i]);
+        Strcpy(buf, materialnm[i]);
+        Strcat(buf, " ");
+        if (!strncmp(dbase_str, buf, ++l)) {
```

Comment:
Can we not have the increment of a variable in the function call? Just because it appears all over NetHack code doesn't mean it's a good practice.